web/client/components/geostory/navigation/Navigation.jsx

```diff
@@ -105,6 +106,7 @@ export default ({
                             <ScrollMenu
                                 items={navigableItems}
                                 currentPage={currentPage}
+                                key={geostoryId}
```

Comment:
@jstrR Even if I suggested you to try to use the key to force the re-render based on story id, I think it's not enough.

This solve the point number three visually on the menu items but following our previous discussion, you pointed out that there is still this [updateCurrentPage](https://github.com/geosolutions-it/MapStore2/blob/f1b1326e41bde2ff29c7383aefa4c81fc58df76f/web/client/components/geostory/layouts/sections/enhancers/currentPageSectionManager.jsx#L43) that read the wrong id from the state.

So the issue is related the fact that the currentStory state is not cleared when a different story is selected in the same session. 

We should:

- [ ] remove this `key={geostoryId}` and the connection of `geostoryId` prop in the related plugin
- [ ] clear the currentStory state in the `loadGeostoryEpic`. @offtherailz suggested to add `.startWith(setCurrentStory({}))` at this [position](https://github.com/geosolutions-it/MapStore2/blob/dddf0be5c42051271cac6d54e179eb1969fe85e0/web/client/epics/geostory.js#L360) 

In this way we ensure that the new state is cleared with the setCurrentStory equal to empty object before the new story request.

Please try the suggested changes and let me know if this solve both: the error on the navbar and the ids used in updateCurrentPage.